pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go

```diff
@@ -200,14 +200,14 @@ func NewAuthCreds(refHostname string) (AuthCreds, error) {
 			// - authConfigHostname:       "https://index.docker.io/v1/" (registry.IndexServer)
 			// - ac.ServerAddress:         "https://index.docker.io/v1/".
 			if !isAuthConfigEmpty(ac) {
-				if authConfigHostname == registry.IndexServer {
-					if ac.ServerAddress != registry.IndexServer {
-						return nil, fmt.Errorf("expected ac.ServerAddress (%q) to be %q", ac.ServerAddress, registry.IndexServer)
-					}
-				} else if ac.ServerAddress == "" {
+				if ac.ServerAddress == "" {
```

Comment:
Order seems to be important here; can you please write unit-tests so that regressions do not occur.